src/nrnpython/nrnpy_p2h.cpp

```diff
@@ -523,24 +523,22 @@ static double guigetval(Object* ho) {
 static void guisetval(Object* ho, double x) {
     PyObject* po = ((Py2Nrn*) ho->u.this_pointer)->po_;
     nanobind::gil_scoped_acquire lock{};
-    PyObject* pn = PyFloat_FromDouble(x);
+    auto pn = nb::steal(PyFloat_FromDouble(x));
     PyObject* p = PyTuple_GetItem(po, 0);
     if (PySequence_Check(p) || PyMapping_Check(p)) {
-        PyObject_SetItem(p, PyTuple_GetItem(po, 1), pn);
+        PyObject_SetItem(p, PyTuple_GetItem(po, 1), pn.ptr());
     } else {
-        PyObject_SetAttr(p, PyTuple_GetItem(po, 1), pn);
+        PyObject_SetAttr(p, PyTuple_GetItem(po, 1), pn.ptr());
```

Comment:
Are we sure this function: https://docs.python.org/3/c-api/object.html#c.PyObject_SetAttr
Raise Ref ?

Reply:
We're not, because the documentation doesn't say. There's this:

> Few functions steal references; the two notable exceptions are [PyList_SetItem()](https://docs.python.org/3/c-api/list.html#c.PyList_SetItem) and [PyTuple_SetItem()](https://docs.python.org/3/c-api/tuple.html#c.PyTuple_SetItem), which steal a reference to the item (but not to the tuple or list into which the item is put!).

https://docs.python.org/3/c-api/intro.html#reference-count-details

and the we hope that not stating anything means it borrows the reference. Searching didn't help. We could write unit-tests that assert the assumptions we make about the Python functions we use (then we also know it doesn't change from version to version).